crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -888,6 +889,73 @@ fn handle_leading_function_with_decorators_comment(comment: DecoratedComment) ->
     }
 }
 
+/// Handles comments between `**` and the variable name in dict unpacking
+/// It attaches these to the appropriate value node
+///
+/// ```python
+/// {
+///     **  # comment between `**` and the variable name
+///     value
+///     ...
+/// }
+/// ```
+fn handle_dict_unpacking_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    match comment.enclosing_node() {
+        // TODO: can maybe also add AnyNodeRef::Arguments here, but tricky to test due to
+        // https://github.com/astral-sh/ruff/issues/5176
+        AnyNodeRef::ExprDict(_) => {}
+        _ => {
+            return CommentPlacement::Default(comment);
+        }
+    };
+    if let Some(following) = comment.following_node() {
+        let preceding_end = match comment.preceding_node() {
+            Some(preceding) => preceding.end(),
+            None => comment.enclosing_node().start(),
+        };
+        let mut tokens_before = SimpleTokenizer::new(
+            locator.contents(),
+            TextRange::new(preceding_end, comment.slice().start()),
+        )
+        .skip_trivia()
+        .peekable();
+
+        while let Some(it) = tokens_before.next() {
+            if it.kind == TokenKind::Star
+                && tokens_before.peek().map(|t| t.kind) == Some(TokenKind::Star)
+            {
+                return CommentPlacement::trailing(following, comment);
+            }
+        }
+    }
+
+    if let Some(preceding) = comment.preceding_node() {
+        let following_start = match comment.following_node() {
+            Some(following) => following.start(),
+            None => comment.enclosing_node().end(),
+        };
+        let mut tokens_after = SimpleTokenizer::new(
+            locator.contents(),
+            TextRange::new(comment.slice().end(), following_start),
+        )
+        .skip_trivia()
+        .peekable();
+
+        while let Some(it) = tokens_after.next() {
+            if it.kind == TokenKind::Star
+                && tokens_after.peek().map(|t| t.kind) == Some(TokenKind::Star)
+            {
+                return CommentPlacement::trailing(preceding, comment);
+            }
+        }
+    }
+
```

Comment:
all my test cases seemed to be handled correctly by the default behaviour for this. i suspect this is being overly cautious because i wasn't sure if that behaviour could be changing in the future. but i think i agree it's better to skip this for now